mosdef_cassandra/tests/test_writers.py

```diff
@@ -1557,3 +1557,77 @@ def test_cbmc_info(self, onecomp_system, twobox_system):
                 cbmc_kappa_dih=5,
                 cbmc_rcut=[],
             )
+
+    @pytest.mark.parametrize('typ,value', [('slitpore', 1), ('cylinder', 1), ('sphere',
+        1), ('interface', [1,2])])
+    def test_write_restricted_gcmc(self, gcmc_system, typ, value):
+        (system, moves) = gcmc_system
+        moves.restricted_type = [[None, typ]]
+        moves.restricted_value = [[None, value]]
+        inp_data = generate_input(
+            system=system,
+            moves=moves,
+            run_type="equilibration",
+            run_length=500,
+            temperature=300.0,
+            chemical_potentials=["none", 10.0],
+        )
+        
+        if typ == 'interface':
+            assert "\nrestricted_insertion {} {} {}\n".format(typ, value[0], value[1]) in inp_data
+        else:
+            assert "\nrestricted_insertion {} {}\n".format(typ, value) in inp_data
+
+    @pytest.mark.parametrize('typ,value', [('slitpore', 3), ('cylinder', 3), ('sphere',
+        3), ('interface', [3,5])])
+    def test_fail_restricted_gcmc(self, gcmc_system, typ, value):
+        (system, moves) = gcmc_system
+        moves.restricted_type = [[None, typ]]
+        moves.restricted_value = [[None, value]]
+        with pytest.raises(ValueError, match=r"Restricted insertion"):
+            inp_data = generate_input(
+                system=system,
+                moves=moves,
+                run_type="equilibration",
+                run_length=500,
+                temperature=300.0,
+                chemical_potentials=["none", 10.0],
+            )
+
+    @pytest.mark.parametrize('typ,value', [('slitpore', 1), ('cylinder', 1), ('sphere',
+        1), ('interface', [1,2])])
+    def test_write_restricted_gemc_npt(self, twocomptwobox_system, typ, value):
+        (system, moves) = twocomptwobox_system
+        moves.restricted_type = [[None, None], [None, typ]]
+        moves.restricted_value = [[None, None], [None, value]]
+        inp_data = generate_input(
+            system=system,
+            moves=moves,
+            run_type="equilibration",
+            run_length=500,
+            temperature=300.0,
+            pressure=1,
+            chemical_potentials=["none", 10.0],
```

Comment:
We don't need to specify a chemical potential for GEMC. In fact we should add an error if someone tries for future.